spotbugsTestCases/src/java/ghIssues/Issue1867.java

```diff
@@ -0,0 +1,15 @@
+package ghIssues;
+
+import java.util.function.Function;
+
+public class Issue1867 {
+	private final Function<String, String> function;
+
+	public Issue1867() {
+		// MC_OVERRIDABLE_METHOD_CALL_IN_CONSTRUCTOR false positive: Claims "append" is overridable
+		function = v -> v + "!";
+
+		// just to ignore warnings about unused member
+		System.out.println(function.apply("hello"));
```

Comment:
My understanding is that this detectors tries to catch a class of vulnerabilities where an attacker extends a class an override a method. For instance you'd override `isAdmin()`, now if the constructor does `this.admin = isAdmin()` the object ends up in an invalid state.
In that particular case `StringBuilder.append()` is not controllable by the attacker, so the bug was a false positive for another reason.

Although this PR seems to fix #1867 (because in the sample the lambda was immediately assigned to a field) the underlying problem with lambdas is still there.

Reply:
Thank you for your explanation.